HIP-6.md

```diff
@@ -0,0 +1,147 @@
+| hip | title                   | status | author | created    |
+| --- | ----------------------- | ------ | ------ | ---------- |
+| 6   | Validator announcements | Draft  | asaj   | 2022-12-21 |
+
+### **Brief Summary / Abstract**
+
+Defines an optional specification for Hyperlane validators to announce themselves and the location of their signed checkpoints.
+
+### **Motivation**
+
+Hyperlane relayers need to aggregate validator signed checkpoints in order to deliver interchain messages.
+
+At present, relayers are manually configured with a list of validators and the location of their signed checkpoints.
+
+Relying on manual configuration creates friction to the introduction of additional validators via sovereign consensus.
+
+This HIP defines a standard for validators to announce themselves to relayers via a smart contract on the underlying blockchain.
+
+Relayers should query this contract to discover the location of validator signed checkpoints.
+
+### **Tech Spec**
+
+#### Announcment attestations
+
+The validator announcement protocol relies on validators signing attestations with the following semantic meaning:
+
+> "Checkpoints for the mailbox contract with address _m_ on chain _d_, signed by the validator with address _v_, can be found at location _l_
+
+The Hyperlane validator binary should sign the following data upon startup.
+
+```
+/**
+ * @notice Returns the digest that Hyperlane validators should sign to support
+ * the announcement protocol
+ * @param _domain The origin domain of the Mailbox being validated
+ * @param _mailbox The address of the Mailbox being validated, as bytes32
+ * @param _storageType An enum representing the modality used to store
```

Comment:
Could a URL based format with the protocol specifying the type be sufficient?

Reply:
Not necessarily (e.g. local storage). Given that we don't know what the other types may be it doesn't feel prudent to make that sort of assumption.

Reply:
somewhat agree that storageType enum here is not only superfluous but potentially confusing to configure outside of solidity

Reply:
What would you propose then? Just bytes with the leading byte as the type?

Reply:
What would not be sufficient about  a URL for local storage? Imo that feels pretty standard (including the protocol at the beginning like file or S3)

Reply:
I think I've encoded what you're looking for